options.go

```diff
@@ -414,3 +416,39 @@ func (s *Server) RemoveTags(tags ...string) *Server {
 	}
 	return s
 }
+
+// Header registers a header parameter for the route.
+func (s *Server) Param(name, description string, params ...OpenAPIParam) *Server {
+	param := GroupParams{Name: name, Description: description}
+
+	for _, p := range params {
+		if p.Required {
+			param.Required = p.Required
+		}
+		if p.Example != "" {
+			param.Example = p.Example
+		}
+		if p.Type != "" {
+			param.Type = p.Type
+		}
+	}
+
+	s.params = append(s.params, param)
+
+	return s
+}
+
+func (s *Server) Header(name, description string, params ...OpenAPIParam) *Server {
+	s.Param(name, description, append(params, OpenAPIParam{Type: "header"})...)
+	return s
+}
+
+func (s *Server) Cookie(name, description string, params ...OpenAPIParam) *Server {
+	s.Param(name, description, append(params, OpenAPIParam{Type: "cookie"})...)
+	return s
+}
+
+func (s *Server) Query(name, description string, params ...OpenAPIParam) *Server {
+	s.Param(name, description, append(params, OpenAPIParam{Type: "query"})...)
+	return s
+}
```

Comment:
Adding coms would be really nice here :) 

Documentation near the usage is amazing